calcom/cal.com
# feat: instant meeting browser notifcations

## What does this PR do?



- Fixes https://github.com/calcom/cal.com/issues/16373 (GitHub issue number)
- Fixes CAL-4177(Linear issue number - should be visible at the bottom of the GitHub issue description)


https://github.com/user-attachments/assets/e1959d74-dfc2-437c-ab3b-ad6f0ee944b2





## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?



Testing it locally is a bit difficult as browser notification only work on https domain. There is a nice article [here](https://vercel.com/guides/access-nextjs-localhost-https-certificate-self-signed) on how we can do that

- Create a new public-private key pair required for web push notifications using `npx web-push generate-vapid-keys`.
- Add these keys in .env file `NEXT_PUBLIC_VAPID_PUBLIC_KEY` and `VAPID_PRIVATE_KEY`.
-  Login in to an org account (owner1-dunder@example.com) 
-  execute ngrok http 3000
-  Now open dev console and copy next auth cookies from app.cal.local to ngrok domain (the idea is to be logged in as owner1-dunder@example.com on ngrok domain)
-  Enable Browser Notification on `bookings/upcoming`

https://github.com/user-attachments/assets/2bef76dd-09a0-4109-93cb-70c4ab1a766b

Now instant meeting event setup

- Create any org team event type 
- Enable Instant bookings toggle
- Go to the booking page and click on connect now bottom modal

You should see a notification which when clicked would redirect you to /connect-and-join page



## Checklist
